src/alternative-gas-token/L1ScrollMessengerNonETH.sol

```diff
@@ -0,0 +1,187 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity =0.8.24;
+
+import {AddressUpgradeable} from "@openzeppelin/contracts-upgradeable/utils/AddressUpgradeable.sol";
+
+import {IL1MessageQueue} from "../L1/rollup/IL1MessageQueue.sol";
+import {IScrollChain} from "../L1/rollup/IScrollChain.sol";
+import {L1ScrollMessenger} from "../L1/L1ScrollMessenger.sol";
+import {IMessageDropCallback} from "../libraries/callbacks/IMessageDropCallback.sol";
+import {ScrollConstants} from "../libraries/constants/ScrollConstants.sol";
+import {WithdrawTrieVerifier} from "../libraries/verifier/WithdrawTrieVerifier.sol";
+
+contract L1ScrollMessengerNonETH is L1ScrollMessenger {
+    /**********
+     * Errors *
+     **********/
+
+    error ErrorDuplicatedMessage();
+
+    error ErrorNonZeroValueFromCaller();
+
+    error ErrorInsufficientMsgValue();
+
+    /*************
+     * Constants *
+     *************/
+
+    /// @notice The address of `L1NativeTokenGateway` contract.
+    address public immutable nativeTokenGateway;
+
+    /***************
+     * Constructor *
+     ***************/
+
+    constructor(
+        address _nativeTokenGateway,
+        address _counterpart,
+        address _rollup,
+        address _messageQueue
+    ) L1ScrollMessenger(_counterpart, _rollup, _messageQueue) {
+        nativeTokenGateway = _nativeTokenGateway;
+    }
+
+    /**********************
+     * Internal Functions *
+     **********************/
+
+    /// @inheritdoc L1ScrollMessenger
+    function _sendMessage(
+        address _to,
+        uint256 _value,
```

Comment:
I think it's a bit confusing to read this contract. Maybe easier to understand if we rename like this.
```suggestion
        uint256 _l2GasTokenValue,
```